### PR TITLE
[BottomNavigation] Fix bug when `setItems` does not correctly set items

### LINKED ISSUE #3177: [BottomNavigation] `setItems` does not correctly set `itemViews` (b/77258452)
### Overview
Reproduced from internal issue.

MDCBottomNavigationBar doesn't display views if items is set twice.

It needs self.itemViews = [NSMutableArray array]; on this line
https://github.com/material-components/material-components-ios/blob/develop/components/BottomNavigation/src/MDCBottomNavigationBar.m#L406

In particular, the itemViews grows each time you call this, leaving the old views.  This causes the real itemViews to be scrolled off the right edge.

### Version number
49.0.0

### PR DESCRIPTION
**Pull request by codeman7**
We might want to use [NSMutableArray array]; instead of `removeAllObjects`.

Closes #3177

### CODE REVIEW

**Review by romoore — CHANGES_REQUESTED** [review]
Thanks for the fix, @codeman7 . Could you please add unit tests for the change? You may need to redeclare `itemViews` in a category for the tests.

**Review by randallli — COMMENTED** [review]
test comment

**Comment by codeman7** [comment]
I don't have very much experience with test, I hope this is what your looking for @romoore.

File: components/BottomNavigation/src/MDCBottomNavigationBar.m [hunk 1]
```diff
@@ -408,15 +408,14 @@ - (void)setItems:(NSArray<UITabBarItem *> *)items {
   if (_items == items) {
     return;
   }
-
```
Comment (romoore) [inline components/BottomNavigation/src/MDCBottomNavigationBar.m]: Nit: Consider restoring this newline and the one ~before~ below?

**Review by romoore — APPROVED** [review]
Yep, looks good to me with just a small nit-pick. Thanks!

File: components/BottomNavigation/tests/unit/BottomNavigationTests.m [hunk 2]
```diff
@@ -93,7 +93,8 @@ -(void)testItemReset {
   self.bottomNavBar.items = @[item1, item2, item3];
 
   // Then
-  XCTAssertEqual(self.bottomNavBar.itemViews.count, 3);
+  unsigned int tabsCount = 3;
```
Comment (romoore) [inline components/BottomNavigation/tests/unit/BottomNavigationTests.m]: Please use NSUInteger, since `unsigned long` is used on 64-bit hosts.

**Review by ianegordon — APPROVED** [review]
LGTM

**Pull request merged**